packages/pwa/app/components/product-tile/index.jsx

```diff
@@ -118,7 +121,12 @@ const ProductTile = (props) => {
             <Text {...styles.title}>{productName}</Text>
 
             {/* Price */}
-            <Text {...styles.price}>{intl.formatNumber(price, {style: 'currency', currency})}</Text>
+            <Text {...styles.price}>
+                {intl.formatNumber(price, {
+                    style: 'currency',
+                    currency: currency || activeCurrency
```

Comment:
```suggestion
                    currency: activeCurrency
```
What if we simplify it to just `activeCurrency`? I see there are other places in the pwa with similar code:
- Some have `currency: currency || activeCurrency`
- while others have `currency: currency || basket.currency || activeCurrency`

Initially I thought all those places should have the same consistent code. But then I wondered whether `currency`, `basket.currency`, `activeCurrency` all have the same value. To me, it looks like they do have the same value:

- `currency` comes from product data that we get through our Commerce API client. We [initialized the client](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/4fe9f3a753bd7a73001350318a4d65509a1f65c0/packages/pwa/app/components/_app-config/index.jsx#L69) with locale and currency. This locale is determined by looking at the page url.
- `basket.currency` comes from the [currency](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/4fe9f3a753bd7a73001350318a4d65509a1f65c0/packages/pwa/app/components/_app/index.jsx#L90) that is also associated with the locale found in the page url (via `useLocale`)
- and finally `activeCurrency` comes from `useCurrency()`/[`CurrencyProvider`](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/4fe9f3a753bd7a73001350318a4d65509a1f65c0/packages/pwa/app/components/_app/index.jsx#L173), which gets the same currency as the basket one above

So all 3 sources above are from the locale found in the page url. Once the locale is found, then the preferred currency can be determined.

Reply:
Another question: how can we make sure to avoid the same bug in the future?

I don't think we can rely on future developers to remember having a fallback currency. How about we tweak the existing hook `useCurrency` to also return `formatPrice`? For example:

```jsx
const {formatPrice} = useCurrency()
return <Text>{formatPrice(price)}</Text>
```

Future developers can call `formatPrice` and not have to worry about fallback currency anymore.

Reply:
Thanks @vmarta for all the detailed investigation. I agree with you, we need to revisit the default currency used in all the components. 

I disagree in one thing the currency is the response from the Einstein API, whereas the activeCurrency from another API response. In theory, they should be the same as you detailed, but the value comes from different API calls.

Let's create a ticket to refactor the default currency in the PWA and merge for now this pattern that follows the pattern used today in other components.